src/osd/scrubber/pg_scrubber.cc

```diff
@@ -820,15 +821,36 @@ void PgScrubber::select_range_n_notify()
   if (select_range()) {
     // the next chunk to handle is not blocked
     dout(20) << __func__ << ": selection OK" << dendl;
-    m_osds->queue_scrub_chunk_free(m_pg, Scrub::scrub_prio_t::low_priority);
-
+    auto cost = get_scrub_cost(m_num_objects, m_is_deep);
+    m_osds->queue_scrub_chunk_free(m_pg, Scrub::scrub_prio_t::low_priority, cost);
   } else {
     // we will wait for the objects range to become available for scrubbing
     dout(10) << __func__ << ": selected chunk is busy" << dendl;
     m_osds->queue_scrub_chunk_busy(m_pg, Scrub::scrub_prio_t::low_priority);
   }
 }
 
+
+uint64_t PgScrubber::get_scrub_cost(int chunk_size, bool is_deep)
+{
+		const auto& conf = m_pg->get_cct()->_conf;
+		uint64_t cost = 0;
+    uint64_t scrub_event_cost = conf->osd_scrub_event_cost;
+    double scrub_metadata_cost = m_osds->get_cost_per_io();
+    if (is_deep) {
+      auto object_size = std::max<uint64_t>(
+      m_pg->info.stats.stats.sum.get_average_object_size(),
+      1);
+      cost = scrub_event_cost + (chunk_size
+      * (scrub_metadata_cost + object_size));
+      dout(20) << __func__ << ": deep-scrub cost = " << cost << dendl;
+			return cost;
```

Comment:
Spacing.  Please make a point of carefully reading over your PR in github prior to asking for a review.